client/src/model/slices/lead_data_slice.ts

```diff
@@ -0,0 +1,51 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { ApiClient } from '@/api/api_client';
+import { AppDispatch } from '@/model/store';
+import { LeadData } from '@/model/lead_data';
+import { LeadDataState } from '@/model/states/lead_data_state';
+
+const initialState: LeadDataState = {};
+const client = new ApiClient();
+
+// Automatically generates action creators and action types that correspond
+// to the reducers and state. See: https://redux-toolkit.js.org/api/createslice
+// TODO(breuch): Consider showing user message on error.
+const leadDataSlice = createSlice({
+  name: 'leadDataSlice',
+  initialState,
+  reducers: {
+    waterSystemQueried(state: LeadDataState, action: PayloadAction<LeadData>) {
```

Comment:
will these contain separate logic in the future? seems like they could be combined to a updateStateWithLeadData action instead?

Reply:
We can add action types to the actions and do one reducer with a switch statement. Let me look into this and do a fast-follow